plugins/homekit/src/types/common.ts

```diff
@@ -96,24 +96,161 @@ export function addCarbonDioxideSensor(device: ScryptedDevice & CO2Sensor, acces
+            mode: value ? mode : HumidityMode.Off
+        });
+    });
+
+    bindCharacteristic(device, ScryptedInterface.HumiditySensor, service, Characteristic.CurrentRelativeHumidity,
+        () => device.humidity);
+
+    bindCharacteristic(device, ScryptedInterface.HumiditySetting, service, Characteristic.CurrentHumidifierDehumidifierState,
+        () => currentState(device.humiditySetting?.activeMode));
+
+    bindCharacteristic(device, ScryptedInterface.HumiditySetting, service, Characteristic.TargetHumidifierDehumidifierState,
+        () => targetState(device.humiditySetting?.mode));
+
+    service.getCharacteristic(Characteristic.TargetHumidifierDehumidifierState).on(CharacteristicEventTypes.SET, (value, callback) => {
+        callback();
+        device.setHumidity({
+            mode: value === Characteristic.TargetHumidifierDehumidifierState.HUMIDIFIER
+                ? HumidityMode.Humidify
+                : value === Characteristic.TargetHumidifierDehumidifierState.DEHUMIDIFIER
+                    ? HumidityMode.Dehumidify
+                    : HumidityMode.Auto
+        });
+    });
+
+    function targetHumidity(setting: HumiditySettingStatus) {
+        if (!setting)
+            return 0;
+
+        if (setting?.availableModes.includes(HumidityMode.Humidify) 
+            && setting?.availableModes.includes(HumidityMode.Dehumidify)) {
+            if (setting?.activeMode === HumidityMode.Humidify)
+                return setting?.humidifierSetpoint;
+            if (setting?.activeMode === HumidityMode.Dehumidify)
+                return setting?.dehumidifierSetpoint;
+
+            return 0;
+        }
+
+        if (setting?.availableModes.includes(HumidityMode.Humidify))
+            return setting?.humidifierSetpoint;
+
+        if (setting?.availableModes.includes(HumidityMode.Dehumidify))
+            return setting?.dehumidifierSetpoint;
+
+        return 0;
+    }
+
+    bindCharacteristic(device, ScryptedInterface.HumiditySetting, service, Characteristic.TargetRelativeHumidity,
+        () => targetHumidity(device.humiditySetting));
+
+    return service;
+}
+
+export function addHumidifier(device: ScryptedDevice & HumiditySetting & HumiditySensor, accessory: Accessory): Service {
+    var service = commonHumidifierDehumidifier(HumidityMode.Humidify, "humidifier", device.name + " Humidifier", device, accessory);
+
+    bindCharacteristic(device, ScryptedInterface.HumiditySetting, service, Characteristic.RelativeHumidityHumidifierThreshold,
+        () => device.humiditySetting?.humidifierSetpoint);
+    service.getCharacteristic(Characteristic.RelativeHumidityHumidifierThreshold).on(CharacteristicEventTypes.SET, (value, callback) => {
+        callback();
+        device.setHumidity({
+            humidifierSetpoint: value as number,
+        });
+    });
+
+    return service;
+}
+
+export function addDehumidifer(device: ScryptedDevice & HumiditySetting & HumiditySensor, accessory: Accessory): Service {
```

Comment:
I don't think this needs export

Reply:
Circling back to this PR, since I missed the original comments... would you like me to submit a new PR for removing the `export`?